react-backend/client/src/components/visitors/qrcode.js

```diff
@@ -53,58 +63,72 @@ export class QRCode extends Component {
       username: '',
       qrcode: '',
       activity: 'not selected',
-      activities: []
-    }
+      activities: [],
+    };
 
     this.handleVideo = this.handleVideo.bind(this);
 
     this.changeActivity = this.changeActivity.bind(this);
   }
 
-  handleVideo = (e) => {
+  handleVideo = e => {
     let newState = {};
-    newState['login'] = this.state.login+1;
+    newState['login'] = this.state.login + 1;
     this.setState(newState);
   };
 
-  changeActivity = (newActivity) => {
+  changeActivity = newActivity => {
     this.setState({
-      activity: newActivity
+      activity: newActivity,
     });
 
     const visitInfo = {
       hash: this.state.hash,
-      activity: newActivity
-    }
+      activity: newActivity,
+    };
 
     return fetch('/postActivity', {
-      method: "POST",
-      body: JSON.stringify(visitInfo)
-    }).then(
-      this.props.history.push('/visitor/end')
-    );
+      method: 'POST',
+      body: JSON.stringify(visitInfo),
+    })
+      .then(this.props.history.push('/visitor/end'))
+      .catch(error => {
+        console.log('ERROR HAPPENING AT FETCH /postActivity', error);
+      });
   };
 
-  componentDidMount(){
-    if(this.state.login===1){
+  componentDidMount() {
+    if (this.state.login === 1) {
       instascan()
-      .then(getUserFromQRScan)
-      .then((user)=>{
-        this.setState({username:user.fullname.fullname, hash:user.fullname.hash});
-      })
+        .then(content => {
+          this.handleVideo();
+          return getUserFromQRScan(content);
+        })
+        .then(user => {
+          this.setState({
+            username: user.fullname.fullname,
+            hash: user.fullname.hash,
+          });
+        })
+        .catch(error => {
```

Comment:
This error is going to be catching everything that could go wrong inside of the instascan functions (anything that rejects or throws in there is going to hit this catch) so I would suggest doing something more meaningful than just a `console.log`, perhaps route the user to `/visitor/qrerror`?